lib/ProductOpener/Ingredients.pm

```diff
@@ -694,7 +696,7 @@ sub parse_ingredients_text($) {
 		my $vegetarian = undef;
 		my $processing = '';
 
-		#print STDERR "s: $s\n";
+		$debug_ingredients and print STDERR "s: $s\n";
```

Comment:
Why don't we use log?

Reply:
I need a way to easily turn on and off debugging messages for a given function, or at least module, for instance for tests in t.

In my .t file I can put something like:
use Log::Any::Adapter 'TAP', filter => 'debug';

But then it turns on debug error messages for all modules and functions, while I just want to see the function I'm working on.

There are probably better ways to do it than what I did. Would you know any?

Reply:
Changed to statements like 
$debug_ingredients and $log->debug("analyze_ingredients_function", { string => $s }) if $log->is_debug();